### DISCUSSION TITLE
Could we clean old gems with updates ?

### DISCUSSION THREAD

**Discussion by Epy** [mastodon/mastodon#18685] [General]
Hello
I'm not used to Ruby applications, so please excuse me if I'm saying something wrong.
My instance is on a small VPS which is getting full each day if I don't run some CLI commands to delete 1 day old medias. I only have a 20GB space, including system which I tried to make the smallest possible, rotating logs etc.
I'd love a view to check how old are my media cache files and where the most comes from for example.

But I also saw that gems and Ruby versions seems to stay on the drive, and for each Ruby version.
For example I have:
```
/home/mastodon/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems
/home/mastodon/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/gems
/home/mastodon/.rbenv/versions/3.0.3/lib/ruby/gems/3.0.0/gems
```
with multiple versions in each !

I'd like to have a command to optionaly delete old gems and old ruby versions when they are not used anymore by the app. It could be a file with a list of current versions which are in use and option to delete all the others or to keep X old versions and delete the others.
I think it could be written in the upgrade procedure.
Thanks

(I wasn't sure if this would be a feature request in a issue or would be a discussion before to see what's the better, sorry if I made a mistake)

**Comment by Justinzobel** [comment]
There are also leftovers in (and cached items that can likely be cleaned):
- /home/mastodon/vendor/bundle/ruby
- /home/mastodon/.yarn/berry/cache
- /home/mastodon/.npm/_cacache
- /home/mastodon/.cache/yarn
- /home/mastodon/.cache/node-gyp
- /home/mastodon/.cache/node
- /home/mastodon/live/node_modules/.cache

**Comment by dmathieu** [comment]
The [bundle clean](https://bundler.io/v2.0/man/bundle-clean.1.html) command will do that for you (only for gems, not for yarn dependencies or other cache folders).
